### PR TITLE
[.NET] Use collection expressions in docs

### PR DESCRIPTION
**Pull request by raulsntos**
As of C# 12 we can now use collection expressions to reduce some boilerplate when initializing collections.

- Related PR: https://github.com/godotengine/godot-docs/pull/10433

### CODE REVIEW

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -17,14 +17,14 @@
 		print(array[-3]) # Prints "Second"
 		[/gdscript]
 		[csharp]
-		var array = new Godot.Collections.Array{"First", 2, 3, "Last"};
+		Godot.Collections.Array array = ["First", 2, 3, "Last"];
 		GD.Print(array[0]); // Prints "First"
 		GD.Print(array[2]); // Prints 3
-		GD.Print(array[array.Count - 1]); // Prints "Last"
+		GD.Print(array[^1]); // Prints "Last"
```
Comment (Mickeon) [inline doc/classes/Array.xml:23]: Is this common knowledge? As a not particularly savvy C# developer, I don't find this shorthand to be more understandable at a glance. Of course, code style is up to you though.

**Comment by AThousandShips** [comment]
> As of C# 12 

Not being familiar with C# versioning but what is the minimum version we support? Is C# 12 expected to be what everyone uses, just so that any code in the docs will work for everyone

File: doc/classes/Array.xml (hunk 1 above)
Comment (raulsntos) [reply doc/classes/Array.xml:23]: I would say so, it's at the very least equivalent to GDScript's negative indexes, which I don't find any easier to understand.

But I imagine the reason why the GDScript example uses negative indexes is to teach users that they can use them to access elements from the end of the array. In a similar way, we can use the C# example to teach them how to do the same thing in C#. So it probably doesn't matter if they're not familiar with this syntax.

**Comment by raulsntos** [comment]
Yes, .NET 8 defaults to C# 12, which is the new minimum required since Godot 4.4.

**Review by mhilbrunner — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**